sandbox/apparmor/notify/listener/listener.go

```diff
@@ -531,9 +536,16 @@ func (l *Listener) newRequest(msg notify.MsgNotificationGeneric) (*Request, erro
 	if err != nil {
 		return nil, err
 	}
+	pid := int(msg.PID())
+	pgid, err := unixGetpgid(pid)
```

Comment:
process group ID can be changed, either by the process itself, or by another process as long they are part of the same session. Fetching this information here feels like TOCTOU kind of issue. By the time we process the request the group may have changed, e.g. app has spawned some multi process batch job and moved it to separate process group such that it can then kill the whole group with `kill(-<group>, SIGKILL)` (not an unusual practice). I would be fine with this if the group id was be included in the message, otherwise this feels a bit racy.